app/main.py

```diff
@@ -1,25 +1,140 @@
-from abc import ABC
+from __future__ import annotations
+from abc import ABC, abstractmethod
 
 
 class IntegerRange:
-    pass
+    def __init__(
+            self,
+            min_amount: int,
+            max_amount: int
+    ) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(
+            self,
+            owner: SlideLimitationValidator,
+            name: str
+    ) -> None:
+        self.private_name = "_" + name
+        self.public_name = name
+
+    def __get__(
+            self,
+            instance: SlideLimitationValidator,
+            owner: Slide
```

Comment:
don't fully get why do we have to implement 'object' as a type annotations for 'owner'. is it because an instance of Any class is inherited from 'object' class?